apis/python/notebooks/tutorial_spatial.ipynb

```diff
@@ -0,0 +1,727 @@
+    "url_base = 'https://cf.10xgenomics.com/samples/spatial-exp/2.1.0'\n",
+    "dataset_name = 'CytAssist_FFPE_Protein_Expression_Human_Glioblastoma'\n",
+    "filtered_h5 = 'CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_filtered_feature_bc_matrix.h5'\n",
+    "spatial_tar = 'CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_spatial.tar.gz'\n",
+    "\n",
+    "# Local download/ingestion paths/configs:\n",
+    "data_root = 'data'     # Sync 10x Visium spatial data into f\"{data_root}/{dataset_name}/10x\". By default, spatial SOMA data will be ingested into f\"{data_root}/{dataset_name}/soma\" as well.\n",
+    "exp_uri = None         # Ingest spatial SOMA data here; defaults to f\"{data_root}/{dataset_name}/soma\"\n",
+    "scene_name = \"scene1\"  # Scene name to write, in ingested spatial SOMA\n",
+    "overwrite_10x = False  # If  already exists, remove and re-ingest it\n",
+    "overwrite_exp = False  # If `exp_uri` already exists, remove and re-ingest it"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "3c210965-0abe-479f-a354-df921d7e64a3",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'data/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma/soma'"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Set default paths\n",
+    "dataset_root = join(data_root, dataset_name)\n",
+    "data_dir_10x = join(dataset_root, '10x')\n",
+    "if exp_uri is None:\n",
+    "    exp_uri = join(dataset_root, 'soma')\n",
+    "exp_uri"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "1437667c-dbee-4f5c-b622-a0842733d227",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def sh(*cmd):\n",
+    "    err(f\"Running: {shlex.join(cmd)}\")\n",
+    "    check_call(cmd)    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2686c818-e180-481a-86ec-6a0772d3019f",
+   "metadata": {},
+   "source": [
+    "### Download sample data from 10x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "e27b622a-a1d0-40c2-bea7-cc9a9bc80596",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "paths = {\n",
+    "    filtered_h5: 'filtered_feature_bc_matrix.h5',\n",
+    "    spatial_tar: spatial_tar\n",
+    "}\n",
+    "for src_name, dst_name in paths.items():\n",
+    "    src = f'{url_base}/{dataset_name}/{src_name}'\n",
+    "    dst = join(data_dir_10x, dst_name)\n",
+    "    if exists(dst):\n",
+    "        if overwrite_10x:\n",
+    "            err(f\"{dst} exists, removing\")\n",
+    "            remove(dst)\n",
+    "    if not exists(dst):\n",
+    "        makedirs(data_dir_10x, exist_ok=True)\n",
+    "        sh('wget', '-qO', dst, src)\n",
+    "        if dst.endswith('.tar.gz'):\n",
+    "            sh('tar', '-C', data_dir_10x, '-xvf', dst)"
```

Comment:
Should this also be skipped if `exp_uri` exists and is not being over-written?

Reply:
I guess the use case is:
1. Run nb for first time, it downloads 10x data, and creates `exp_uri`
2. Clean up local dir by removing the 10x data, which is no longer needed
3. Run nb subsequent times, it will re-download 10x data

Makes sense to me to skip downloading 10x data in case 3., as you suggest, let me make that change…